libchisel/src/lib.rs

```diff
@@ -35,6 +40,10 @@ pub trait ModuleValidator {
     fn validate(&self, module: &Module) -> Result<bool, ModuleError>;
 }
 
+pub trait InstructionValidator {
+    fn validate(&mut self, module: &Module) -> Result<bool, InstructionError>;
```

Comment:
I think since we have `ModuleError`, it should be possible to merge the `InstructionError` enum into it and thus removing the need for a new trait.